src/Components/CreateImageWizardV2/steps/TargetEnvironment/Azure/AzureSourcesSelect.tsx

```diff
@@ -31,51 +39,45 @@ const AzureSourcesSelect = ({ label, isRequired, className, ...props }) => {
     isError,
     refetch,
   } = useGetSourceListQuery({ provider: 'azure' });
-  const sources = extractProvisioningList(rawSources);
 
   const {
     data: sourceDetails,
     isFetching: isFetchingDetails,
     isSuccess: isSuccessDetails,
     isError: isErrorDetails,
   } = useGetSourceUploadInfoQuery(
-    { id: selectedSourceId },
+    // @ts-ignore
+    { id: azureSource },
     {
-      skip: !selectedSourceId,
+      skip: !azureSource,
     }
   );
+  tenantId = sourceDetails?.azure?.tenant_id || '';
```

Comment:
This causes us to lose the subscription created earlier when `tenantId` was initialized using the `useAppSelector` hook. Additionally, assigning a value to `tenantId` like this does not result in an update to the value in the store. Instead, we should be using `useAppDispatch` to update the `tenantId` value in the Redux store.